qiskit_experiments/data_processing/nodes.py

```diff
@@ -0,0 +1,204 @@
+            average: if True the single-shots are averaged.
+        """
+        self.scale = scale
+        self.average = average
+        super().__init__()
+        self._accepted_inputs = ["memory"]
+
+    @abstractmethod
+    def _index(self) -> int:
+        """Return 0 for real and 1 for imaginary part."""
+
+    def process(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Modifies the data inplace by taking the real part of the memory and
+        scaling it by the given factor.
+
+        Args:
+            data: The data dict. IQ data is stored under memory.
+
+        Returns:
+            processed data: A dict with the data.
+
+        Raises:
+            DataProcessorError: if the data does not contain memory.
+        """
+
+        # Single shot data
+        if isinstance(data["memory"][0][0], list):
+            new_mem = []
+            for shot in data["memory"]:
+                new_mem.append([self.scale * iq_qubit[self._index()] for iq_qubit in shot])
+
+            if self.average:
+                new_mem = list(np.mean(np.array(new_mem), axis=0))
+
+        # Averaged data
+        else:
+            new_mem = [self.scale * iq_qubit[self._index] for iq_qubit in data["memory"]]
+
+        return {self.node_output: new_mem}
+
+
+class ToReal(IQPart):
+    """IQ data post-processing. Isolate the real part of the IQ data."""
+
+    @property
+    def node_output(self) -> str:
+        """Key under which ToReal stores the data."""
+        return "memory_real"
+
+    def _index(self) -> int:
+        """Return 0 for real part."""
+        return 0
+
+
+class ToImag(IQPart):
+    """IQ data post-processing. Isolate the imaginary part of the IQ data."""
+
+    @property
+    def node_output(self) -> str:
+        """Key under which ToImag stores the data."""
+        return "memory_imag"
+
+    def _index(self) -> int:
+        """Return 0 for real part."""
+        return 1
+
+
+class Population(DataAction):
+    """Count data post processing. This returns population."""
+
+    def __init__(self):
+        """Initialize a counts to population data conversion."""
+        super().__init__()
+        self._accepted_inputs = ["counts"]
+
+    @property
+    def node_output(self) -> str:
+        """Key under which Population stores the data."""
+        return "populations"
+
+    def process(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Args:
+            data: The data dictionary. This will modify the dict in place,
+                taking the data under counts and adding the corresponding
+                populations.
+
+        Returns:
+            processed data: A dict with the populations.
+
+        Raises:
+            DataProcessorError: if counts are not in the given data.
+        """
+
+        counts = data.get("counts")
+
+        populations = np.zeros(len(list(counts.keys())[0]))
+
+        shots = 0
```

Comment:
Can we get shots from metadata? I think it's stored in the experimental configuration.

Reply:
Perhaps, but summing the counts is more robust and does not cost much to do. So I will leave this as is.